app/ide-desktop/lib/cloud/src/authentication/src/api.tsx

```
@@ -0,0 +1,116 @@
+
+
+
+// =============
+// === Types ===
+// =============
+
+interface Request {
+    accessToken: string;
+    path: string;
+    method: HttpMethod;
+    body?: Record<string, string>;
+}
+
+enum HttpMethod {
+    get = "GET",
+    post = "POST",
+    put = "PUT",
+    delete = "DELETE",
+}
+
+export type Organization = {
+    id: string;
+    userEmail: string;
+    name: string;
+}
+
+
+
+// ================
+// === FetchApi ===
+// ================
+
+/// Utility function for performing a REST request to an API endpoint.
+///
+/// Parses the response as JSON (if possible) and returns the result.
+/// Otherwise returns `null` on a successful API call.
+const fetchApi = async (
+    {
+        accessToken,
+        path,
+        method,
+        body,
+    }: Request,
+): Promise<Response> => {
+    if (!accessToken) { throw new Error("No access token provided"); }
+    
+    const config: RequestInit = { method };
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const headers = new Headers(config.headers);
+    headers.set("Authorization", `Bearer ${accessToken}`);
+    if (body != null) {
+        config.body = JSON.stringify(body);
+        headers.set("Content-Type", "application/json");
+    }
+
+    config.headers = headers;
+    const response = await fetch(`${API_URL}/${path}`, config);
+
+    return response;
+}
+
+
+
+// ===================
+// === SetUsername ===
+// ===================
+
+export type SetUsernameBody = {
+    userName: string;
+    userEmail: string;
+};
+
+/// Wrapper for our `setUsername` Cloud backend API endpoint.
+export const setUsername = async (accessToken: string, body: SetUsernameBody) => {
+    const path = "users";
+    const method = HttpMethod.post;
+    const request = { accessToken, path, method, body };
+    const response = await fetchApi(request);
+    const text = await response.text();
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```

do not disable such checks! This is really bad and can hide errors